src/test/java/org/broadinstitute/hellbender/utils/config/ConfigUnitTest.java

```diff
@@ -0,0 +1,507 @@
+        ConfigUtils.checkFileNamePropertyExistenceAndSetConfigFactoryProperties(sourcesFilenames);
+
+        // Now we check them:
+        for ( int i = 0 ; i < sourcesFilenames.size() ; ++i ) {
+
+            // If we added them to the properties, we don't need to set the names in
+            // the ConfigFactory.  Therefore they should be null:
+            if ( addToSystemProperties.get(i) ) {
+                Assert.assertEquals(ConfigFactory.getProperty(sourcesFilenames.get(i)), null);
+            }
+            // If the filename didn't exist in the system properties, then we did need to add them to the
+            // ConfigFactory properties:
+            else {
+                Assert.assertEquals(ConfigFactory.getProperty(sourcesFilenames.get(i)), ConfigUtils.NO_PATH_VARIABLE_VALUE);
+            }
+        }
+
+        // Remove these from our System Properties and ConfigFactory properties:
+        for ( int i = 0 ; i < sourcesFilenames.size() ; ++i ) {
+            if ( addToSystemProperties.get(i) ) {
+                System.clearProperty(sourcesFilenames.get(i));
+            }
+            else {
+                ConfigFactory.clearProperty(sourcesFilenames.get(i));
+                Assert.assertEquals(ConfigFactory.getProperty(sourcesFilenames.get(i)), null);
+            }
+        }
+    }
+
+    @Test(dataProvider= "createConfigListsAndResults")
+    void testConfigUtilsGetConfigPathVariableNamesFromConfigClasses(final List<Class<?>> configurationClasses, final List<String> expectedFilePathVariables ) {
+        Assert.assertEquals( ConfigUtils.getConfigPathVariableNamesFromConfigClasses(configurationClasses), expectedFilePathVariables );
+    }
+
+    @Test(dataProvider = "createConfigAndExpectedProperties")
+    <T extends Config> void testConfigUtilsGetSystemPropertiesFromConfig(final T config, final Map<String, String> expectedProperties) {
+        Assert.assertEquals( ConfigUtils.getSystemPropertiesFromConfig(config), expectedProperties );
+    }
+
+    @Test(dataProvider = "createPropertiesToInjectIntoSystem")
+    void testConfigUtilsInjectToSystemProperties(final Map<String, String> propertiesToInject) {
+
+        // Ensure our properties are not there to start with:
+        for ( final String key : propertiesToInject.keySet() ) {
+            Assert.assertEquals(System.getProperty(key), null);
+        }
+
+        // Inject the stuff:
+        ConfigUtils.injectToSystemProperties(propertiesToInject);
+
+        // Verify our injection:
+        for ( final String key : propertiesToInject.keySet() ) {
+            Assert.assertEquals(System.getProperty(key), propertiesToInject.get(key));
+        }
+
+        // Reset our properties after:
+        for ( final String key : propertiesToInject.keySet() ) {
+            System.clearProperty(key);
+            Assert.assertEquals(System.getProperty(key), null);
+        }
+    }
+
+    @Test
+    void testDerivedClassProperties() throws IOException {
```

Comment:
No prob.  I realized I didn't have it when I saw your comment about derived types.